packages/flutter/lib/src/widgets/scroll_position.dart

```diff
@@ -489,11 +489,20 @@ abstract class ScrollPosition extends ViewportOffset with ScrollMetrics {
       assert(minScrollExtent != null);
       assert(maxScrollExtent != null);
       assert(minScrollExtent <= maxScrollExtent);
+      final bool isFirstApplication = _minScrollExtent == null || _maxScrollExtent == null;
       _minScrollExtent = minScrollExtent;
       _maxScrollExtent = maxScrollExtent;
       _haveDimensions = true;
-      applyNewDimensions();
       _didChangeViewportDimensionOrReceiveCorrection = false;
+      if (!isFirstApplication) {
```

Comment:
In the course of trying to get this working there are a dozen tests that assume you can start a ScrollPosition out of bounds.  I think it has no ramifications in practical use, but keeps tests from breakinig.